torchx/distributed/__init__.py

```diff
@@ -0,0 +1,261 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Convenience methods to use ``torch.distributed``.
+"""
+
+import logging
+import os
+import warnings
+from contextlib import contextmanager
+from typing import Any, Iterator, Literal
+
+import torch
+import torch.distributed as dist
+
+log: logging.Logger = logging.getLogger(__name__)
+
+
+def local_rank() -> int:
+    """
+    Returns the local rank (aka rank within the node) of this process.
+    Typically the local rank is used to set the CUDA device on the node.
+
+    .. warning::
+        This function only works correctly if the invoker of the program sets ``LOCAL_RANK`` env var
+        or invokes the program with ``torchrun`` (aka ``torch.distributed.run``) or ``torchx``.
+        If ``LOCAL_RANK`` is not set or the process group is not initialized
+        then this function assumes that the process is not distributed and trivially returns 0.
+
+    """
+
+    if not dist.is_initialized():
+        return 0
+
+    if "LOCAL_RANK" not in os.environ:
+        warnings.warn(
+            "\n"
+            "==============================================================================================\n"
+            "`LOCAL_RANK` environment variable is not set. Will trivially return 0 for local_rank.\n"
+            " It is recommended to use torchrun/torchx to run your script or set the `LOCAL_RANK` manually.\n"
+            " For additional details see:\n"
+            "  1) https://pytorch.org/torchx/latest/components/distributed.html\n"
+            "  2) https://pytorch.org/docs/stable/elastic/run.html\n"
+            "=============================================================================================="
+        )
+        return 0
+    else:
+        return int(os.environ["LOCAL_RANK"])
```

Comment:
torch distributed doesn't provide a helper, huh

Reply:
yeah... I thought there was as `dist.get_local_rank()` too... probaby should upstream this function to `dist_c10d.py`